operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/DependentResourceConfiguration.java

```diff
@@ -0,0 +1,10 @@
+package io.javaoperatorsdk.operator.api.config;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+public interface DependentResourceConfiguration<R, P extends HasMetadata> {
```

Comment:
Why are these classes configurable? or what is the point of this configuration?

Reply:
The goal is to decouple the configuration of the dependent resource from its creation. Previously, this was done by the same class and the `AnnotationConfiguration` was doing the instantiating of the `DependentResource` implementation which led to it being too complex but also needed to know some details about how the implementation worked (in particular how to deal with Kubernetes dependent resources).
This also forced the quarkus extension to have to instantiate the dependent resource implementation class at build time because it was right there on the `ControllerConfiguration.getDependentResources()` method. Well, it's not exactly true because it's possible to delay the instantiation until the method is called but that wasn't clean enough and led to more complications.